source/cgame/cg_players.cpp

```diff
@@ -267,4 +269,20 @@ void CG_LoadClientInfo( cg_clientInfo_t *ci, const char *info, int client )
 
 	s = Info_ValueForKey( info, "m" );
 	ci->modelindex = s && s[0] ? atoi( s ) : 0;
+
+
+	s = Info_ValueForKey( info, "steam_id" );
+	if (s && atol(s)){
+		ci->steamid = atoll(s);
+		CGAME_IMPORT.Steam_RequestAvatar(ci->steamid, 0);
+	}
+}
+void CG_CallbackRequestAvatar(uint64_t steamid, char *avatar){
+	for (int i = 0; i < gs.maxclients; i++){
+		cg_clientInfo_t *ci = &cgs.clientInfo[i];
+		if (ci->steamid == steamid){
+			ci->avatar = trap_R_RegisterRawPic(va("avatar-%llu", ci->steamid), 32, 32, (uint8_t*)avatar, 4);
```

Comment:
I am not sure if this picture also should be unregistered again?

Reply:
how?

Reply:
seems to be handled automatically then, that makes it easier.